### PR TITLE
Add scheduler to full finetuning and adapter scripts

### LINKED ISSUE #406: Adding an optional cosine scheduler
We might want to add a cosine scheduler and enable it via a command-line arg. It's pretty much standard for LLM pretraining an finetuning.

### PR DESCRIPTION
**Pull request by rasbt**
Adds learning rate schedulers to the full finetuning script as well as the Adapter scripts. It's similar to what I added to LoRA last week.

Fixes #406

### CODE REVIEW

**Comment by carmocca** [comment]
@awaelchli Would you like to review and land this?

**Review by awaelchli — APPROVED** [review]
